crowbar_framework/app/controllers/installer_controller.rb

```diff
@@ -0,0 +1,147 @@
+# limitations under the License.
+#
+
+class InstallerController < ApplicationController
+  skip_before_filter :enforce_installer
+  before_filter :hide_navigation
+  after_filter :post_install
+
+  def index
+    @steps = all_steps
+  end
+
+  #
+  # Perform Crowbar Installation
+  #
+  # Provides the restful api call for
+  # /installer/status 	GET 	return done steps, error and success
+  # returns a hash with an indicator if the installation failed/succeeded and the steps that are done
+  def status
+    respond_to do |format|
+      format.json { render json: steps_done }
+      format.html { redirect_to installer_url }
+    end
+  end
+
+  #
+  # Set the status of an installation step
+  #
+  # Provides the restful api call for
+  # /installer/update 	POST 	set the status
+  # touch a file in /var/lib/crowbar named after the step
+  # not really needed yet, but maybe later
+  def update
+    step = params[:step]
+    return render_not_found unless all_steps.include? step.to_sym
+    write_file(crowbar_lib_dir, step)
+    respond_to do |format|
+      format.json { head :ok }
+      format.html { redirect_to installer_url }
+    end
+  end
+
+  #
+  # Perform Crowbar Installation
+  #
+  # Provides the restful api call for
+  # /installer/install 	POST 	triggers install-chef-suse.sh
+  # the bash Process will be spawned in the background and therefore has not a direct return value which we can use here
+  def install
+    installed_ok = Rails.root.join(".crowbar-installed-ok")
+    crowbar_dir = Rails.root.join("..")
+    if installed_ok.exist?
+      msg = "If you want to reinstall please remove /opt/dell/crowbar_framework/crowbar-installed-ok"
+      respond_to do |format|
+        format.json { render text: msg and return }
+        format.html { redirect_to installer_url and return }
+      end
+    end
+
+    if File.exist?("#{crowbar_lib_dir.to_path}/crowbar_installing")
+      flash[:notice] = "Crowbar is already installing. Please wait."
+    else
+      pid = spawn("sudo #{crowbar_dir}/bin/install-chef-suse.sh --crowbar")
+      write_file(crowbar_lib_dir, "crowbar_installing")
+      Process.detach(pid)
+    end
+
+    respond_to do |format|
+      format.json { head :ok }
+      format.html { redirect_to installer_url }
+    end
+  end
+
+  protected
+
+  def hide_navigation
+    @hide_navigation = true
+  end
+
+  def all_steps
+    [
+      :pre_sanity_checks,
+      :run_services,
+      :initial_chef_client,
+      :barclamp_install,
+      :bootstrap_crowbar_setup,
+      :apply_crowbar_config,
+      :transition_crowbar,
+      :chef_client_daemon,
+      :post_sanity_checks
+    ]
+  end
+
+  def crowbar_lib_dir
+    Pathname.new("/var/lib/crowbar")
+  end
+
+  def steps_done
+    steps = all_steps.select do |step|
+      crowbar_lib_dir.join(step.to_s).exist?
+    end
+    { steps: steps, failed: failed?, done: successful?, error: error_msg, success: success_msg, notice: notice_msg }
```

Comment:
Metrics/LineLength: Line is too long. [116/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)